src/messages.ts

```diff
@@ -543,20 +549,77 @@ export class Messages<T extends string> {
     exitCodeOrCause?: number | Error,
     cause?: Error
   ): SfdxError {
+    const structuredMessage = this.formatMessageContents('error', key, tokens, actionTokens);
+    return new SfdxError(
+      structuredMessage.message,
+      structuredMessage.name,
+      structuredMessage.actions,
+      exitCodeOrCause,
+      cause
+    );
+  }
+
+  /**
+   * Convenience method to create warning using message labels.
+   *
+   * `warning.name` will be the upper-cased key, remove prefixed `warning.` and will always end in Warning.
+   * `warning.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   */
+  public createWarning(key: T, tokens: Tokens = [], actionTokens: Tokens = []): StructuredMessage {
+    return this.formatMessageContents('warning', key, tokens, actionTokens);
+  }
+
+  /**
+   * Convenience method to create info using message labels.
+   *
+   * `info.name` will be the upper-cased key, remove prefixed `info.` and will always end in Info.
+   * `info.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   */
+  public createInfo(key: T, tokens: Tokens = [], actionTokens: Tokens = []): StructuredMessage {
+    return this.formatMessageContents('info', key, tokens, actionTokens);
+  }
+
+  /**
+   * Formats message contents given a message type, key, tokens and actions tokens
+   *
+   * `<type>.name` will be the upper-cased key, remove prefixed `<type>.` and will always end in 'Error | Warning | Info.
+   * `<type>.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param type The type of the message set must 'error' | 'warning' | 'info'.
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   */
+  private formatMessageContents(
+    type: 'error' | 'warning' | 'info',
+    key: T,
+    tokens: Tokens = [],
+    actionTokens: Tokens = []
+  ): StructuredMessage {
+    const label = upperFirst(type);
+    const labelRegExp = new RegExp(`${label}$`);
+    const searchValue: RegExp = type === 'error' ? /^error.*\./ : new RegExp(`^${type}.`);
     // Convert key to name:
-    //     'myMessage' -> `MyMessageError`
-    //     'myMessageError' -> `MyMessageError`
-    //     'error.myMessage' -> `MyMessageError`
-    //     'errors.myMessage' -> `MyMessageError`
-    const errName = `${upperFirst(key.replace(/^errors*\./, ''))}${/Error$/.exec(key) ? '' : 'Error'}`;
-    const errMessage = this.getMessage(key, tokens);
-    let errActions;
+    //     'myMessage' -> `MyMessageWarning`
+    //     'myMessageError' -> `MyMessageWarning`
```

Comment:
Is this comment accurate or should it be this instead?
```suggestion
    //     'myMessageError' -> `MyMessageError`
```